### PR TITLE
les: fix announce receiving

### PR DESCRIPTION
**Pull request by b00ris**
Closes https://github.com/status-im/go-ethereum/issues/55

### CODE REVIEW

File: les/peer.go [hunk 1]
```diff
@@ -423,11 +423,11 @@ func (p *peer) Handshake(td *big.Int, head common.Hash, headNum uint64, genesis
 		p.fcCosts = list.decode()
 	} else {
 		//on client node
-		p.announceType = announceTypeSimple
+		p.requestAnnounceType = announceTypeSimple
 		if p.isTrusted {
-			p.announceType = uint64(announceTypeSigned)
+			p.requestAnnounceType = uint64(announceTypeSigned)
```
Comment (themue) [inline les/peer.go]: Question here: Why are the `requestAnnounceType` and the `announceTypeSimple` of type `uint64` while the `announceTypeSigned` has to be casted?

File: les/ulc_test.go [hunk 2]
```diff
@@ -40,6 +40,48 @@ func TestULCSyncWithOnePeer(t *testing.T) {
...
+func TestULCSyncWithOnePeer1(t *testing.T) {
+	f := newFullPeerPair(t, 1, 4, testChainGen)
+	ulcConfig := &eth.ULCConfig{
+		MinTrustedFraction: 100,
+		TrustedNodes:       []string{f.ID.String()},
+	}
+
+	l := newLightPeer(t, ulcConfig)
+
+	if reflect.DeepEqual(f.PM.blockchain.CurrentHeader().Hash(), l.PM.blockchain.CurrentHeader().Hash()) {
```
Comment (themue) [inline les/ulc_test.go]: Why this check? Is there any chance that these new and different instances share the same header? If yes I also would use a more detailed message, e.g. "header blocks of full and light peer are equal".

File: les/ulc_test.go [hunk 3]
```diff
@@ -40,6 +40,48 @@ func TestULCSyncWithOnePeer(t *testing.T) {
...
+
+	fPeer, lPeer, err := connectPeers(f, l, 2)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	l.PM.synchronise(fPeer)
+
+	if !reflect.DeepEqual(f.PM.blockchain.CurrentHeader().Hash(), l.PM.blockchain.CurrentHeader().Hash()) {
+		t.Fatal("sync don't work")
```
Comment (themue) [inline les/ulc_test.go]: "sync doesn't work"

File: les/ulc_test.go [hunk 4]
```diff
@@ -40,6 +40,48 @@ func TestULCSyncWithOnePeer(t *testing.T) {
...
+	}
+
+	l.PM.peers.lock.Lock()
+	fmt.Println(len(l.PM.peers.peers))
```
Comment (themue) [inline les/ulc_test.go]: Better `t.Logf()`.

**Comment by b00ris** [comment]
@themue Fixed

File: les/ulc_test.go [hunk 5]
```diff
@@ -36,11 +37,55 @@ func TestULCSyncWithOnePeer(t *testing.T) {
...
+	}
+
+	l.PM.synchronise(fPeer)
+
+	//check that the sync is finished correctly
+	if !reflect.DeepEqual(f.PM.blockchain.CurrentHeader().Hash(), l.PM.blockchain.CurrentHeader().Hash()) {
+		t.Fatal("sync doesn't work")
+	}
+
+	l.PM.peers.lock.Lock()
```
Comment (JekaMas) [inline les/ulc_test.go]: Can we add something like 'l.PM.peersCount()' and remove all locks?

**Pull request merged**